client/blocks/subscription-length-picker/index.jsx

```diff
@@ -0,0 +1,193 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal Dependencies
+ */
+import { getPlansBySiteId, isRequestingSitePlans } from 'state/sites/plans/selectors';
+import { localize } from 'i18n-calypso';
+import formatCurrency from 'lib/format-currency';
+import { getProductsList, isProductsListFetching } from 'state/products-list/selectors';
+import { requestProductsList } from 'state/products-list/actions';
+import { requestPlans } from 'state/plans/actions';
+import { getPlanRawPrice, isRequestingPlans } from '../../state/plans/selectors';
+import { getCurrentUserCurrencyCode } from '../../state/current-user/selectors';
+import { getPlanDiscountedRawPrice } from '../../state/sites/plans/selectors';
+import { getSelectedSiteId } from '../../state/ui/selectors';
+import { abtest } from 'lib/abtest';
+import { getPlan, applyTestFiltersToPlansList } from 'lib/plans';
+import { TERM_MONTHLY } from 'lib/plans/constants';
+import SubscriptionLengthOption from './option';
+
+export class SubscriptionLengthPicker extends React.Component {
+	static propTypes = {
+		initialValue: PropTypes.string,
+		currencyCode: PropTypes.string.isRequired,
+		productsWithPrices: PropTypes.array.isRequired,
+		onChange: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
```

Comment:
It's more important to define what folks are expected to pass in versus connected props. (Though the latter is still quite useful.)

```jsx
plans: PropTypes.oneOf( [ PLAN_BUSINESS, PLAN_BUSINESS_2_YEARS ] ),
initialValue: PropTypes.string,
onChange: PropTypes.func,
// connected props
translate: PropTypes.func,
```

Reply:
Note that to test a component by itself, we use the base component vs the wrapped class. So often `translate` is not required in PropTypes.